keps/3125-maximum-execution-time/README.md

```diff
@@ -0,0 +1,210 @@
+# KEP-3125: Workload Maximum Execution Time
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API](#api)
+    - [Workload Spec](#workload-spec)
+    - [Workload Status](#workload-status)
+  - [Constants](#constants)
+  - [Controller](#controller)
+    - [Workload](#workload)
+    - [Jobs / Jobframework](#jobs--jobframework)
+  - [Webhooks](#webhooks)
+    - [Workload](#workload-1)
+    - [Jobs](#jobs)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+      - [controller/core/workload](#controllercoreworkload)
+      - [controller/jobs/job](#controllerjobsjob)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Add the ability to set a maximum execution time of a Job.
+
+## Motivation
+
+Extend Kueue so that kjobctl can incorporate functionalities equivalent to `-t` in [slurm](https://slurm.schedmd.com/).
+
+Different Jobs CRDs have a field with similar semantics, but there is no standard. For example `batch/Job` has `spec.activeDeadlneSeconds`, while JobSet does not have such an API for now.
+
+### Goals
+
+Introduce a Job agnostic API of setting and enforcing the maximum execution time of a Job.
+
+
+### Non-Goals
+
+- Change the behavior of the Kueue's scheduler based on the maximum execution time.
+- Mitigate potential conflicting configurations between `kueue.x-k8s.io/max-exec-time-seconds` and other job specific `activeDeadline` functionalities.
+
+## Proposal
+
+A new Kueue specific label is added `kueue.x-k8s.io/max-exec-time-seconds` containing
+the maximum number of seconds the Job is expected to execute.
+
+The duration is passed to the Job's Workload.
+
+If a workload stays admitted for longer that it is expected it will be Deactivated. 
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a Batch User, I want to have the ability to set a limit on how long my job can run.
+
+### Notes/Constraints/Caveats (Optional)
+
+
+### Risks and Mitigations
+
+## Design Details
```

Comment:
Please make the label immutable.

The reason is that when we use it for slurm-like scheduling the one could abuse the label by setting very small value on admission and extending after admission. 

We could mitigate this by evicting on changes, but it adds complexity so it might be better to revisit mutability for GA. 

cc @tenzen-y @alculquicondor